### DISCUSSION TITLE
Decrease the confidence interval in the Fixed Noise GP model

### DISCUSSION THREAD

**Discussion by sambitmishra98** [meta-pytorch/botorch#1418] [Q&A] [answered]
I have used FixedNoiseGP on 4 datapoints as shown in the figure. Based on my experience with the blac-box, I know that the trend is roughly linear and variance from x=40 to x=100 will be similar to the variance at x=40 or x=100. So I want to decrease the confidence interval at all these points to be smaller. For example, I want the 95% confidence interval at x=80 to be between y=9 and y=11. I don't know how to include this to my model. Can you point me to the argument which I should change for this?

![image](https://user-images.githubusercontent.com/63449209/192030618-82278306-7f56-45f0-8f68-ac329f98f335.png)

I want to do this because I dont want the Expected Improvement acquisition function to later on search for an optimum (minimum) value in the region approximately above x=80.

**Comment by saitcakmak** [comment, marked as answer]
Hi @sambitmishra98. I don't think there's a way to include the domain info into the model out of the box. You would need to define either a custom model or a custom mean and covariance modules to do this.

If you wanted a really hacky solution, you could add a dummy observation between 40 and 80 to tell the model that those areas are not good, or that there's a roughly linear pattern there, by having a roughly linear dummy observations in that region.

>  I dont want the Expected Improvement acquisition function to later on search for an optimum (minimum) value in the region approximately above x=80.

Alternatively, you can achieve this by excluding the regions you know to be inferior from the search space bounds while optimizing the acquisition function.

**Reply by sambitmishra98** [reply]
I didn't mean I want to include domain info, that was a small example for what I wanted. Basically if I have a few datapoints with a certain value of variance, I want the rest of the region to not be too far away from that value. For example, if we have just one datapoint as shown in figure, the model decides that it will not go beyond a certain variance value in the entire domain. The Gaussian distribution at every point in the domain does not keep increasing as we go farther away from the datapoints. So there must be some default variance that the model approaches to if we don't have datapoints around. I hope I am able to communicate this properly.

![image](https://user-images.githubusercontent.com/63449209/192039886-ec46273f-6d8b-4b1c-b7cd-c4d57932c338.png)

**Reply by sambitmishra98** [reply]
To add to my explanation, for a certain GP variance at location x=50 in the above figure, I want the variance to increase more gradually as we sample candidates away from x=50.

**Reply by saitcakmak** [reply]
The posterior variance is fully determined by the `covar_module` (i.e., the prior covariance matrix) and the locations of the observations (`train_x`). During model training, the hyper-parameters of the `covar_module` are set to a value that best describes the observations (using MAP estimation). The well known GP posterior update formula tells us that the prior covariance is an upper bound on the posterior covariance, and the posterior variance will shrink as you get closer to the points of observations, with the shrinkage being more significant the smaller the observation noise level is. 

The default covariance kernel we use is Matern 5/2 with ARD. It also has some length-scale & output-scale priors, which you can find in the code. Your options are to either change the kernel and/or the priors (you can pass a custom `covar_module` while constructing the model), or to modify the hyper-parameters after the model training.

**Reply by sambitmishra98** [reply]
Thanks a lot! This is exactly what I wanted.
